google/resource_dataflow_job.go

```diff
@@ -0,0 +1,210 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"google.golang.org/api/dataflow/v1b3"
+	"google.golang.org/api/googleapi"
+)
+
+var dataflowTerminalStatesMap = map[string]bool{
+	"JOB_STATE_UNKNOWN":    false,
+	"JOB_STATE_STOPPED":    false,
+	"JOB_STATE_RUNNING":    false,
+	"JOB_STATE_DONE":       true,
+	"JOB_STATE_FAILED":     true,
+	"JOB_STATE_CANCELLED":  true,
+	"JOB_STATE_UPDATED":    true,
+	"JOB_STATE_DRAINING":   true,
+	"JOB_STATE_DRAINED":    true,
+	"JOB_STATE_PENDING":    false,
+	"JOB_STATE_CANCELLING": true,
+}
+
+func resourceDataflowJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataflowJobCreate,
+		Read:   resourceDataflowJobRead,
+		Delete: resourceDataflowJobDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"template_gcs_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"temp_gcs_location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"max_workers": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"on_delete": &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"cancel", "drain"}, false),
+				Optional:     true,
+				Default:      "drain",
+				ForceNew:     true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDataflowJobCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	jobName := d.Get("name").(string)
+	gcsPath := d.Get("template_gcs_path").(string)
+	tempLocation := d.Get("temp_gcs_location").(string)
+	zone := d.Get("zone").(string)
+	maxWorkers := d.Get("max_workers").(int)
+	params := expandDataflowParamsStringMap(d.Get("parameters").(map[string]interface{}))
```

Comment:
you can use expandStringMap from utils.go for this

Reply:
Done.